kubectl.fish

```diff
@@ -222,12 +241,13 @@ function __fish_print_resource -d 'Print a list of resources' -a resource
   and return
 
   set -l namespace (__fish_kubectl_get_namespace)
+  set -l context (__fish_kubectl_get_context)
   test -z "$namespace"
   and __fish_kubectl get "$resource" -o name \
     | string replace -r '(.*)/' ''
   and return
 
-  __fish_kubectl --namespace "$namespace" get "$resource" -o name \
+  __fish_kubectl --namespace "$namespace" --context "$context" get "$resource" -o name \
```

Comment:
Should we maybe put `--context "$context"` inside the `__fish_kubectl` function so that it applies everywhere? I think that would make it less likely for us to miss pass the context to things like fetching the custom resource definitions when completing `kubectl get <tab>` and things like that.

Reply:
Thanks. Does that mean the function would look like the following?
```
function __fish_kubectl
  set -l context (__fish_kubectl_get_context)
  command kubectl --context "$context" $__k8s_timeout $argv
end
```
And of course removing the changes to `__fish_print_resource`.

I'm VERY new to fish and this is the only script I've dug into.

Reply:
definitely on the right track here. That function would also need to check that context exists before passing that flag though, because I know not everyone wants to have to pass context to every command.

Reply:
I do believe kubectl has to _have_ a context, though.  If you just have a single cluster and context, that is simply the _default_ context.  

You can see what that context is with `kubectl config current-context`... so you should be able to set the `context` variable to the output of `kubectl config current-context` as a default, to be safe.

Reply:
ah yea, that's a good point.

Reply:
Sorry it's been over 9 months, I honestly forgot about this. I've updated the code to calculate the context within the __fish_kubectl function. There should be no reason to query for the current context, we can just leave it out if it's not on the command line....similar to what's done with the namespace in __fish_kubectl_print_resource when it calls __fish_kubectl_get_ns_flags. I basically copied that function and changed it to work with context.

As I mentioned before, I've never written fish scripts...and it's still true 9 months later. :) So if there's anything I mistakenly assumed, please let me know. Thanks!